canopen/node/remote.py

```diff
@@ -89,27 +89,46 @@ def restore(self, subindex=1):
+        """
+        try:
+            if subindex is not None:
+                logger.info(str('SDO [{index}][{subindex}]: {name}: {value}'.format(
+                    index=format(index, '#04x'),
+                    subindex=subindex,
+                    name=name,
+                    value=format(value, '#04x'))))
+                self.sdo[index][subindex].raw = value
+            else:
+                self.sdo[index].raw = value
+                logger.info(str('SDO [{index}]: {name}: {value}'.format(
+                    index=format(index, '#04x'),
+                    name=name,
+                    value=format(value, '#04x'))))
+        except canopen.SdoCommunicationError as e:
+            logger.info(str(e))
+        except canopen.SdoAbortedError as e:
+            # WORKAROUND for broken implementations: the SDO is set but the error
+            # "Attempt to write a read-only object" is raised any way.
+            if e.code != 0x06010002:
+                # Abort codes other than "Attempt to write a read-only object"
+                # should still be reported.
+                logger.warning('[ERROR SETTING object {0}:{1}]  {2}'.format(index, subindex, str(e)))
+                logger.info(str(e))
+                raise
+
     def load_configuration(self):
         ''' Load the configuration of the node from the object dictionary.'''
         for obj in self.object_dictionary.values():
             if isinstance(obj, Record) or isinstance(obj, Array):
                 for subobj in obj.values():
-                    if isinstance(subobj, Variable) and (subobj.access_type == 'rw') and (subobj.value is not None) :
-                        logger.debug(str('SDO [{index}][{subindex}]: {name}: {value}'.format(
-                            index=subobj.index,
-                            subindex=subobj.subindex,
-                            name=subobj.name,
-                            value=subobj.value)))
-                        try:
-                            self.sdo[subobj.index][subobj.subindex].raw = subobj.value
-                        except canopen.SdoCommunicationError as e:
-                            logger.info(str(e))
-                        except canopen.SdoAbortedError as e:
-                            # WORKAROUND for broken implementations: the SDO is set but the error
-                            # "Attempt to write a read-only object" is raised any way.
-                            if e.code != 0x06010002:
-                                # Abort codes other than "Attempt to write a read-only object"
-                                # should still be reported.
-                                print('[ERROR SETTING object {0}:{1}]  {2}'.format(subobj.index, subobj.subindex, str(e)))
-                                logger.info(str(e))
-                                raise
+                    if isinstance(subobj, Variable) and (subobj.access_type in ('rw', 'rww')) and (subobj.value is not None):
```

Comment:
I've added the `.writable` attribute to Variable as a higher level API than `.access_type`.

Reply:
Ok, it's more clean this way. Done!